broadcast/internal/outter_errors.go

```diff
@@ -0,0 +1,44 @@
+package outter_errors
```

Comment:
- Typically, in Go, the package name is synonymous with the directory name containing the package files.
- Also, the package name `outter_errors` is not idiomatic in Go; package names are generally single words without underscores.

I suggest to move the file to a directory like that:
```
├── internal/
│   └── outter_errors/
│       └── outer_errors.go
```

Reply:
Changed to `WithCause()`,  please re-review